### PR TITLE
First version of core

### PR DESCRIPTION
**Pull request by Tyrannas**
Optimized for sklearn (no pyfunc or keras for now)
Missing tests

### CODE REVIEW

File: mlflow_client/core.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+
+def get_auto_backend():
+    """
+    Returns a backend automatically
+    LocalBackend if mlflow is not found
+    MLFLow Local if mlflow is found
+    MLFlow Server if mlflow is found and MLFLOW_TRACKING_URI defined in env variables
+    """
+    try:
```
Comment (gfournier) [inline mlflow_client/core.py]: add a logger message to display which environment is chosen

File: mlflow_client/core.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        root_path = path if path else os.path.dirname(__file__)
+        self._path = os.path.join(root_path, 'mlruns')
+        self._run_started = 0
+
+    @if_run_active
+    def log_metric(self, metric: str, value: int):
+        path = os.path.join(self._path, str(self._run_started), 'metrics', metric)
+        os.makedirs(os.path.dirname(path), exist_ok=True)
+        with open(path, 'w') as f:
+            f.write(str(value))
```
Comment (gfournier) [inline mlflow_client/core.py]: log in JSON format, better if we build some tools to reaggregate all files of a local folder.

```
{ "name": metric, "value": value }
```

File: mlflow_client/core.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+    @if_run_active
+    def log_parameter(self, parameter: str, value):
+        path = os.path.join(self._path, str(self._run_started), 'parameters', parameter)
```
Comment (gfournier) [inline mlflow_client/core.py]: same as above

File: mlflow_client/core.py [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+    @if_run_active
+    def log_model(self, model, name: str = None, library: MLFrameworks = MLFrameworks.PYTHON):
+        if not name:
+            name = str(uuid.uuid1()) + '.pkl'
+
+        path = os.path.join(self._path, str(self._run_started), 'artifacts', 'model', name)
+        os.makedirs(os.path.dirname(path), exist_ok=True)
+
+        packages_info = ""
```
Comment (gfournier) [inline mlflow_client/core.py]: Maybe save also a `pip show` output ?

**Review by gfournier — CHANGES_REQUESTED** [review]

File: mlflow_client/core.py (hunk 2 above)
Comment (Tyrannas) [reply mlflow_client/core.py]: I did {"metricName": [metricValues]} instead

File: mlflow_client/core.py (hunk 1 above)
Comment (Tyrannas) [reply mlflow_client/core.py]: i put a logging warning so that every logger with default config could log this message, should I put it to info ?

**Pull request merged**